config/tools/comparator/main.go

```diff
@@ -0,0 +1,103 @@
+package main
+
+import (
+	_ "embed"
+	"encoding/json"
+	"fmt"
+	"golang.org/x/text/cases"
+	"golang.org/x/text/language"
+	"os"
+	"strings"
+
+	"github.com/crossplane/upjet/pkg/registry"
+	"github.com/pkg/errors"
+	"github.com/scaleway/provider-scaleway/config/tools"
+
+	"gopkg.in/yaml.v3"
+)
+
+// ProviderMetadata represents the structure of provider-metadata.yaml
+type ProviderMetadata struct {
+	Resources map[string]*registry.Resource `yaml:"resources"`
+}
+
+func main() {
+	currentProviderMetadata := os.Getenv("CURRENT_METADATA")
+	newProviderMetadata := os.Getenv("NEW_METADATA")
+
+	currentResources, err := parseProviderMetadata(currentProviderMetadata)
+	if err != nil {
+		fmt.Printf("Error parsing current provider metadata: %v\n", err)
+		return
+	}
+
+	newResources, err := parseProviderMetadata(newProviderMetadata)
+	if err != nil {
+		fmt.Printf("Error parsing new provider metadata: %v\n", err)
+		return
+	}
+
+	addedResources := findNewResources(currentResources, newResources)
+	fmt.Println("New resources found:")
+	resourceConfigs := make([]tools.ResourceConfig, 0, len(addedResources))
+
+	for _, resource := range addedResources {
+		references := make(map[string]string)
+
+		for _, example := range resource.Examples {
+			for refKey, refValue := range example.References {
+				resourceType := strings.Split(refValue, ".")[0]
+				references[refKey] = resourceType
+			}
+		}
+
+		config := tools.ResourceConfig{
+			PackageName:           strings.ToLower(strings.ReplaceAll(resource.SubCategory, " ", "")), // Convert "Apple Silicon" to "applesilicon"
```

Comment:
Could you extract it and include some tests for this function? I'm betting that it will be probably from this piece of code that we will have unexpected behavior in some code generation.